File: Farm/Farm.UI/Farm/src/app/Models/actionable-task.ts
export interface ActionableTask {
  id: string;

  taskType: string;// Irrigation, fertilizing, Harvesting

  farmName: string;

  fieldName: string;

  cropName: string;

  taskTime: string | Date;

  isCompleted: boolean;

  createdAt?: string;

  updatedAt?: string;
}

File: Farm/Farm.Domain/Entities/ActionableTask.cs
using Farm.Domain.Common;

namespace Farm.Domain.Entities
{
    public class ActionableTask : BaseEntity
    {
        public ActionableTask()
        {
            IsCompleted = false;
        }

        public string TaskType { get; set; }//Irrigation, fertilizing, Harvesting

        public string FarmName { get; set; }

        public string FieldName { get; set; }

        public string CropName { get; set; }

        public DateTime? TaskTime { get; set; }

        public bool IsCompleted { get; set; }

        public Field Field { get; set; }
        public Guid FieldId { get; set; }
    }
}
